---
title: "Fine-mapping with summary statistics"
author: "Yuxin Zou and Gao Wang"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Fine-mapping with summary statistics}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse = TRUE,comment = "#",fig.width = 5,
                      fig.height = 3,fig.align = "center",
                      fig.cap = "&nbsp;",dpi = 120)
```

This vignette demonstrates how to use `susieR` with "summary statistics" in the context of genetic fine-mapping. 
We use the same simulated data as in [fine mapping vignette](finemapping.html). The simulated data is expression level of a gene ($y$) in $N \approx 600$ individuals. 
We want to identify with the genotype matrix $X_{N\times P}$ ($P=1000$) the genetic variables that causes changes in expression level. The data-set is shipped with `susieR`. It is simulated to have exactly 3 non-zero effects.

```{r}
library(susieR)
set.seed(1)
```

## The data-set

```{r}
data(N3finemapping)
attach(N3finemapping)
```

Notice that we've simulated 2 sets of $Y$ as 2 simulation replicates. Here we'll focus on the first data-set.

```{r}
dim(data$Y)
```

Here are the 3 "true" signals in the first data-set:

```{r}
b <- data$true_coef[,1]
plot(b, pch=16, ylab='effect size')
```

```{r}
which(b != 0)
```

So the underlying causal variables are 403, 653 and 773.

## Summary statistics from simple regression

Summary statistics of genetic association studies typically contain effect size ($\hat{\beta}$ coefficient from regression), p-value and minor allele frequencies. 
These information can be used to perform fine-mapping with given an additional input of correlation matrix between variables.
The correlation matrix in genetics is typically referred to as LD matrix (LD for linkage disequilibrium). 
One may use external reference panels to estimate it when this matrix cannot be obtained from samples directly. There is a potential problem using LD from reference panel, and we demonstrated it in the next section.
Caution that LD matrix here has to be correlation matrix $r$, not $r^2$ or $abs(r)$.

Our example data-set additionally provides summary statistics for fitting per-variable univariate simple regression.
The results are $\hat{\beta}$ and $SE(\hat{\beta})$ from which z-scores can be derived.
Alternatively you can obtain z-scores from $\hat{\beta}$ and p-values if you are provided with those information.
Again we focus only on results from the first data-set:

```{r}
z_scores <- sumstats[1,,] / sumstats[2,,]
z_scores <- z_scores[,1]
susie_plot(z_scores, y = "z", b=b)
```

For this example the correlation matrix can be computed directly from data provide,

```{r}
R <- cor(data$X)
```

## Fine-mapping with `susieR` using summary statistics

For starters, we assume there are at most 10 causal variables, i.e. set `L = 10`, although SuSiE is generally robust to the choice of `L`.

SuSiE regression with summary statistics is implemented as `susie_rss` function, 

```{r}
fitted_rss <- susie_rss(z_scores, R, L = 10,
                          estimate_residual_variance = TRUE, 
                          estimate_prior_variance = TRUE)
```

Using `summary` function, we can examine the posterior inclusion probability (PIP) for each variable, and the 95% credible sets. 

Here, we are the 95% credible sets.

```{r}
summary(fitted_rss)$cs
```

The 3 causal signals have been captured by the 3 CS reported here. The
3rd CS contains many variables, including the true causal variable
403.

We can also plot the posterior inclusion probability (PIP),

```{r}
susie_plot(fitted_rss, y="PIP", b=b)
```

The true causal variables are colored red. The 95% CS identified are circled in different colors.

The result should be very similar to using the individual level data for this example, as the z-scores and correlation matrix (LD reference)
come from the same genotype data-set.

```{r, fig.width=6,fig.height=6}
fitted = susie(data$X, data$Y[,1], L = 10,
                estimate_residual_variance = TRUE, 
                estimate_prior_variance = TRUE)
plot(fitted$pip, fitted_rss$pip, ylim=c(0,1))
```

## Using LD from reference panel

When original genotype information is not available, one may use reference panel to estimate LD matrix. If the LD pattern from the reference panel does not agree well with the original sample, it does not agree with the observed z scores neither. There could be lots of false discoveries. 

We randomly generated 500 samples from $N(0,R)$ and treated them as reference panel genotype matrix `X_ref`.
```{r echo=F}
set.seed(1234)
tmp = matrix(rnorm(500*1001), 500, 1001)
eigenR = eigen(R)
eigenR$values[eigenR$values < 1e-10] = 0
X_ref = tmp %*% (eigenR$values * t(eigenR$vectors))
R_ref = cor(X_ref)
```

Using susie_rss function with LD from reference panel, the model identifies 10 CSs and only one of them contains the true signal. 
```{r}
fitted_rss_ref <- susie_rss(z_scores, R_ref, L = 10,
                            estimate_residual_variance = TRUE, 
                            estimate_prior_variance = TRUE, check_z = F)
```

```{r}
susie_plot(fitted_rss_ref, y="PIP", b=b)
```

**We recommend the following modification on the LD matrix from reference panel.** Suppose `X_ref` is column-standardized, we recommend estimating LD as
$$
\tilde{R} = \text{cov2cor}(\frac{1}{n_{ref}}(X_{ref}^\intercal X_{ref}+ zz^\intercal)) = \text{cov2cor}(\frac{1}{n_{ref}}(R_{ref}(n_{ref}-1)+ zz^\intercal)),
$$
where z is the original summary statistics, $n_{ref}$ is the sample size from the reference panel.

```{r}
R_ref_cor = cov2cor(R_ref*(500-1) + tcrossprod(z_scores))
fitted_rss_ref_cor <- susie_rss(z_scores, R_ref_cor, L = 10,
                                estimate_residual_variance = TRUE, 
                                estimate_prior_variance = TRUE, check_z = F)
```

```{r}
summary(fitted_rss_ref_cor)$cs
```

The model identifies 2 CSes with 2 true signals. It misses 1 true signal, but it is a better fit than using the correlation matrix from the reference pancel directly.

```{r}
susie_plot(fitted_rss_ref_cor, y="PIP", b=b)
```

## Session information

Here are some details about the computing environment, including the
versions of R, and the R packages, used to generate these results.

```{r}
sessionInfo()
```
