<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Fine-mapping with summary statistics • susieR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Fine-mapping with summary statistics">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">susieR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stephenslab/susieR">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Fine-mapping with summary statistics</h1>
                        <h4 class="author">Yuxin Zou and Gao Wang</h4>
            
            <h4 class="date">2020-01-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/stephenslab/susieR/blob/master/vignettes/finemapping_summary_statistics.Rmd"><code>vignettes/finemapping_summary_statistics.Rmd</code></a></small>
      <div class="hidden name"><code>finemapping_summary_statistics.Rmd</code></div>

    </div>

    
    
<p>This vignette demonstrates how to use <code>susieR</code> with “summary statistics” in the context of genetic fine-mapping. We use the same simulated data as in <a href="finemapping.html">fine mapping vignette</a>. The simulated data is expression level of a gene (<span class="math inline">\(y\)</span>) in <span class="math inline">\(N \approx 600\)</span> individuals. We want to identify with the genotype matrix <span class="math inline">\(X_{N\times P}\)</span> (<span class="math inline">\(P=1000\)</span>) the genetic variables that causes changes in expression level. The data-set is shipped with <code>susieR</code>. It is simulated to have exactly 3 non-zero effects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(susieR)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">1</span>)</code></pre></div>
<div id="the-data-set" class="section level2">
<h2 class="hasAnchor">
<a href="#the-data-set" class="anchor"></a>The data-set</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(N3finemapping)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/attach">attach</a></span>(N3finemapping)</code></pre></div>
<p>Notice that we’ve simulated 2 sets of <span class="math inline">\(Y\)</span> as 2 simulation replicates. Here we’ll focus on the first data-set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/dim">dim</a></span>(data$Y)
<span class="co"># [1] 574   2</span></code></pre></div>
<p>Here are the 3 “true” signals in the first data-set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">b &lt;-<span class="st"> </span>data$true_coef[,<span class="dv">1</span>]
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(b, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">ylab=</span><span class="st">'effect size'</span>)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-4-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(b !=<span class="st"> </span><span class="dv">0</span>)
<span class="co"># [1] 403 653 773</span></code></pre></div>
<p>So the underlying causal variables are 403, 653 and 773.</p>
</div>
<div id="summary-statistics-from-simple-regression" class="section level2">
<h2 class="hasAnchor">
<a href="#summary-statistics-from-simple-regression" class="anchor"></a>Summary statistics from simple regression</h2>
<p>Summary statistics of genetic association studies typically contain effect size (<span class="math inline">\(\hat{\beta}\)</span> coefficient from regression), p-value and minor allele frequencies. These information can be used to perform fine-mapping with given an additional input of correlation matrix between variables. The correlation matrix in genetics is typically referred to as LD matrix (LD for linkage disequilibrium). One may use external reference panels to estimate it when this matrix cannot be obtained from samples directly. Caution that LD matrix here has to be correlation matrix <span class="math inline">\(r\)</span>, not <span class="math inline">\(r^2\)</span> or <span class="math inline">\(abs(r)\)</span>.</p>
<p>Our example data-set additionally provides summary statistics for fitting per-variable univariate simple regression. The results are <span class="math inline">\(\hat{\beta}\)</span> and <span class="math inline">\(SE(\hat{\beta})\)</span> from which z-scores can be derived. Alternatively you can obtain z-scores from <span class="math inline">\(\hat{\beta}\)</span> and p-values if you are provided with those information. Again we focus only on results from the first data-set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">z_scores &lt;-<span class="st"> </span>sumstats[<span class="dv">1</span>,,] /<span class="st"> </span>sumstats[<span class="dv">2</span>,,]
z_scores &lt;-<span class="st"> </span>z_scores[,<span class="dv">1</span>]
<span class="kw"><a href="../reference/susie_plot.html">susie_plot</a></span>(z_scores, <span class="dt">y =</span> <span class="st">"z"</span>, <span class="dt">b=</span>b)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-6-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>For this example the correlation matrix can be computed directly from data provide,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">R &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/cor">cor</a></span>(data$X)</code></pre></div>
</div>
<div id="fine-mapping-with-susier-using-summary-statistics" class="section level2">
<h2 class="hasAnchor">
<a href="#fine-mapping-with-susier-using-summary-statistics" class="anchor"></a>Fine-mapping with <code>susieR</code> using summary statistics</h2>
<p>For starters, we assume there are at most 10 causal variables, i.e. set <code>L = 10</code>, although SuSiE is generally robust to the choice of <code>L</code>.</p>
<p>SuSiE regression with summary statistics is implemented as <code>susie_rss</code> function,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitted_rss &lt;-<span class="st"> </span><span class="kw"><a href="../reference/susie_rss.html">susie_rss</a></span>(z_scores, R, <span class="dt">L =</span> <span class="dv">10</span>,
                          <span class="dt">estimate_residual_variance =</span> <span class="ot">TRUE</span>, 
                          <span class="dt">estimate_prior_variance =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Using <code>summary</code> function, we can examine the posterior inclusion probability (PIP) for each variable, and the 95% credible sets.</p>
<p>Here, we are the 95% credible sets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/summary">summary</a></span>(fitted_rss)$cs
<span class="co">#   cs cs_log10bf cs_avg_r2 cs_min_r2</span>
<span class="co"># 1  2   8.415429 1.0000000 1.0000000</span>
<span class="co"># 2  1  14.908636 0.9816575 0.9634847</span>
<span class="co"># 3  3   6.889567 0.9211685 0.6907024</span>
<span class="co">#                                                                                                          variable</span>
<span class="co"># 1                                                                                                             653</span>
<span class="co"># 2                                                                                                         773,777</span>
<span class="co"># 3 360,362,365,372,373,374,379,381,383,384,386,387,388,389,391,392,396,397,398,399,400,401,403,404,405,407,408,415</span></code></pre></div>
<p>The 3 causal signals have been captured by the 3 CS reported here. The 3rd CS contains many variables, including the true causal variable 403.</p>
<p>We can also plot the posterior inclusion probability (PIP),</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/susie_plot.html">susie_plot</a></span>(fitted_rss, <span class="dt">y=</span><span class="st">"PIP"</span>, <span class="dt">b=</span>b)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-10-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>The true causal variables are colored red. The 95% CS identified are circled in different colors.</p>
<p>The result should be very similar to using the individual level data for this example, as the z-scores and correlation matrix (LD reference) come from the same genotype data-set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitted =<span class="st"> </span><span class="kw"><a href="../reference/susie.html">susie</a></span>(data$X, data$Y[,<span class="dv">1</span>], <span class="dt">L =</span> <span class="dv">10</span>,
                <span class="dt">estimate_residual_variance =</span> <span class="ot">TRUE</span>, 
                <span class="dt">estimate_prior_variance =</span> <span class="ot">TRUE</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(fitted$pip, fitted_rss$pip, <span class="dt">ylim=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>,<span class="dv">1</span>))</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-11-1.png" alt="&amp;nbsp;" width="720"><p class="caption">
 
</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<p>Here are some details about the computing environment, including the versions of R, and the R packages, used to generate these results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()
<span class="co"># R version 3.5.3 (2019-03-11)</span>
<span class="co"># Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co"># Running under: BunsenLabs GNU/Linux 8.9 (Hydrogen)</span>
<span class="co"># </span>
<span class="co"># Matrix products: default</span>
<span class="co"># BLAS: /usr/lib/atlas-base/atlas/libblas.so.3.0</span>
<span class="co"># LAPACK: /usr/lib/atlas-base/atlas/liblapack.so.3.0</span>
<span class="co"># </span>
<span class="co"># locale:</span>
<span class="co">#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co"># [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co"># </span>
<span class="co"># attached base packages:</span>
<span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co"># </span>
<span class="co"># other attached packages:</span>
<span class="co"># [1] susieR_0.8.1.0571</span>
<span class="co"># </span>
<span class="co"># loaded via a namespace (and not attached):</span>
<span class="co">#  [1] Rcpp_1.0.3       knitr_1.23       xml2_1.2.0       magrittr_1.5    </span>
<span class="co">#  [5] roxygen2_6.1.1   MASS_7.3-51.1    lattice_0.20-38  R6_2.4.1        </span>
<span class="co">#  [9] rlang_0.4.2      highr_0.8        stringr_1.4.0    tools_3.5.3     </span>
<span class="co"># [13] grid_3.5.3       xfun_0.7         htmltools_0.3.6  commonmark_1.7  </span>
<span class="co"># [17] yaml_2.2.0       digest_0.6.23    assertthat_0.2.1 rprojroot_1.3-2 </span>
<span class="co"># [21] pkgdown_1.3.0    crayon_1.3.4     Matrix_1.2-16    fs_1.2.7        </span>
<span class="co"># [25] memoise_1.1.0    evaluate_0.13    rmarkdown_1.13   stringi_1.4.3   </span>
<span class="co"># [29] compiler_3.5.3   desc_1.2.0       backports_1.1.5</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#the-data-set">The data-set</a></li>
      <li><a href="#summary-statistics-from-simple-regression">Summary statistics from simple regression</a></li>
      <li><a href="#fine-mapping-with-susier-using-summary-statistics">Fine-mapping with <code>susieR</code> using summary statistics</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Gao Wang, Yuxin Zou, Kaiqian Zhang, Peter Carbonetto, Matthew Stephens.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
