<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Fine-mapping examples • susieR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="Fine-mapping examples">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">susieR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stephenslab/susieR">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Fine-mapping examples</h1>
                        <h4 class="author">Gao Wang</h4>
            
            <h4 class="date">2018-09-20</h4>
          </div>

    
    
<div class="contents">
<p>This vignette demonstrates <code>susieR</code> in the context of genetic fine-mapping. We use simulated data of expression level of a gene (<span class="math inline">\(y\)</span>) in <span class="math inline">\(N \approx 600\)</span> individuals. We want to identify with the genotype matrix <span class="math inline">\(X_{N\times P}\)</span> (<span class="math inline">\(P=1000\)</span>) the genetic variables that causes changes in expression level.</p>
<p>The simulated data-set is <a href="https://github.com/stephenslab/susieR/blob/master/inst/datafiles/N3finemapping.rds">available here</a>, as part of the <code>susieR</code> package. It is simulated to have exactly 3 non-zero effects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(susieR)
<span class="kw">set.seed</span>(<span class="dv">1</span>)</code></pre></div>
<div id="the-data-set" class="section level2">
<h2 class="hasAnchor">
<a href="#the-data-set" class="anchor"></a>The data-set</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dat =<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">system.file</span>(<span class="st">"datafiles"</span>,<span class="st">"N3finemapping.rds"</span>,<span class="dt">package =</span> <span class="st">"susieR"</span>))
<span class="kw">names</span>(dat)
<span class="co"># [1] "data"     "sumstats"</span></code></pre></div>
<p><code>data</code> contains regression data-set <span class="math inline">\(X\)</span> and <span class="math inline">\(y\)</span>, along with some other relevant properties in the context of genetic studies. It also contains the “true” regression coefficent the data is simulated from.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(dat<span class="op">$</span>data)
<span class="co"># [1] "X"                 "chrom"             "pos"              </span>
<span class="co"># [4] "true_coef"         "residual_variance" "Y"                </span>
<span class="co"># [7] "allele_freq"       "V"</span></code></pre></div>
<p>Notice that we’ve simulated 2 sets of <span class="math inline">\(Y\)</span> as 2 simulation replicates. Here we’ll focus on the first data-set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(dat<span class="op">$</span>data<span class="op">$</span>Y)
<span class="co"># [1] 574   2</span></code></pre></div>
<p>Here are the 3 “true” signals in the first data-set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">b =<span class="st"> </span>dat<span class="op">$</span>data<span class="op">$</span>true_coef[,<span class="dv">1</span>]
<span class="kw">plot</span>(b, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">ylab=</span><span class="st">'effect size'</span>)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_files/figure-html/unnamed-chunk-5-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">which</span>(b <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>)
<span class="co"># [1] 403 653 773</span></code></pre></div>
<p>So the underlying causal variables are 403, 653 and 773.</p>
</div>
<div id="simple-regression-summary-statistics" class="section level2">
<h2 class="hasAnchor">
<a href="#simple-regression-summary-statistics" class="anchor"></a>Simple regression summary statistics</h2>
<p>The data-set additionally provides summary statistics for fitting univariate simple regression variable by variable. The results are <span class="math inline">\(\hat{\beta}\)</span> and <span class="math inline">\(SE(\hat{\beta})\)</span> from which z-scores can be derived. Again we focus only on results from the first data-set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">z_scores =<span class="st"> </span>dat<span class="op">$</span>sumstats[<span class="dv">1</span>,,] <span class="op">/</span><span class="st"> </span>dat<span class="op">$</span>sumstats[<span class="dv">2</span>,,]
z_scores =<span class="st"> </span>z_scores[,<span class="dv">1</span>]
<span class="kw"><a href="../reference/susie_plot.html">susie_plot</a></span>(z_scores, <span class="dt">y=</span><span class="st">'z'</span>, <span class="dt">b=</span>b)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_files/figure-html/unnamed-chunk-7-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
</div>
<div id="fine-mapping-with-susier" class="section level2">
<h2 class="hasAnchor">
<a href="#fine-mapping-with-susier" class="anchor"></a>Fine-mapping with <code>susieR</code>
</h2>
<p>For starters, we assume there are at most 5 causal variables, i.e., set <code>L=5</code>, although SuSiE is robust to the choice of <code>L</code>.</p>
<p>We set SuSiE prior variance to 0.1, and update residual variance in the variational algorithm that fits SuSiE model. The <code>susieR</code> function call is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitted =<span class="st"> </span><span class="kw"><a href="../reference/susie.html">susie</a></span>(dat<span class="op">$</span>data<span class="op">$</span>X, dat<span class="op">$</span>data<span class="op">$</span>Y[,<span class="dv">1</span>],
               <span class="dt">L=</span><span class="dv">5</span>,
               <span class="dt">estimate_residual_variance =</span> <span class="ot">TRUE</span>, 
               <span class="dt">scaled_prior_variance =</span> <span class="fl">0.1</span>)</code></pre></div>
<div id="confidence-sets" class="section level3">
<h3 class="hasAnchor">
<a href="#confidence-sets" class="anchor"></a>Confidence sets</h3>
<p>By default we output 95% confidence set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(fitted<span class="op">$</span>sets)
<span class="co"># $cs</span>
<span class="co"># $cs$L3</span>
<span class="co"># [1] 653</span>
<span class="co"># </span>
<span class="co"># $cs$L1</span>
<span class="co"># [1] 773 777</span>
<span class="co"># </span>
<span class="co"># $cs$L2</span>
<span class="co">#  [1] 362 372 373 374 379 381 383 384 386 387 388 389 391 392 396 397 398</span>
<span class="co"># [18] 399 400 401 403 404 405 407 408 415</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># $purity</span>
<span class="co">#    min.abs.corr mean.abs.corr median.abs.corr</span>
<span class="co"># L3    1.0000000     1.0000000       1.0000000</span>
<span class="co"># L1    0.9815726     0.9907863       0.9907863</span>
<span class="co"># L2    0.8686309     0.9673674       0.9756251</span>
<span class="co"># </span>
<span class="co"># $cs_index</span>
<span class="co"># [1] 3 1 2</span>
<span class="co"># </span>
<span class="co"># $coverage</span>
<span class="co"># [1] 0.95</span></code></pre></div>
<p>The 3 causal signals have been captured by the 3 CS reported here. The 3rd CS contains many variables, including the true causal variable <code>403</code>. The minimum absolute correlation is 0.86.</p>
<p>If we use the default 90% coverage for confidence sets, we still capture the 3 signals, but “purity” of the 3rd CS is now 0.91 and size of the CS is also a bit smaller.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sets =<span class="st"> </span><span class="kw"><a href="../reference/susie_get_CS.html">susie_get_CS</a></span>(fitted,
                    <span class="dt">X =</span> dat<span class="op">$</span>data<span class="op">$</span>X,
                    <span class="dt">coverage =</span> <span class="fl">0.9</span>,
                    <span class="dt">min_abs_corr =</span> <span class="fl">0.1</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(sets)
<span class="co"># $cs</span>
<span class="co"># $cs$L3</span>
<span class="co"># [1] 653</span>
<span class="co"># </span>
<span class="co"># $cs$L1</span>
<span class="co"># [1] 773 777</span>
<span class="co"># </span>
<span class="co"># $cs$L2</span>
<span class="co">#  [1] 373 374 379 381 383 384 386 387 388 389 391 392 396 399 400 401 403</span>
<span class="co"># [18] 404 405 407 408</span>
<span class="co"># </span>
<span class="co"># </span>
<span class="co"># $purity</span>
<span class="co">#    min.abs.corr mean.abs.corr median.abs.corr</span>
<span class="co"># L3    1.0000000     1.0000000       1.0000000</span>
<span class="co"># L1    0.9815726     0.9907863       0.9907863</span>
<span class="co"># L2    0.9119572     0.9744588       0.9779763</span>
<span class="co"># </span>
<span class="co"># $cs_index</span>
<span class="co"># [1] 3 1 2</span>
<span class="co"># </span>
<span class="co"># $coverage</span>
<span class="co"># [1] 0.9</span></code></pre></div>
</div>
<div id="posterior-inclusion-probability" class="section level3">
<h3 class="hasAnchor">
<a href="#posterior-inclusion-probability" class="anchor"></a>Posterior inclusion probability</h3>
<p>Previously we’ve determined that summing over 3 single effect regression models is approperate for our application. Here we summarize the variable selection results by posterior inclusion probability (PIP):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/susie_plot.html">susie_plot</a></span>(fitted, <span class="dt">y=</span><span class="st">'PIP'</span>, <span class="dt">b=</span>b)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_files/figure-html/unnamed-chunk-12-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>The true causal variables are colored red. The 95% CS identified are circled in different colors. Of interest is the cluster around position 400. The true signal is 403 but apparently it does not have the highest PIP. To compare ranking of PIP and original z-score in that CS:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">z3 =<span class="st"> </span><span class="kw">cbind</span>(fitted<span class="op">$</span>sets<span class="op">$</span>cs[[<span class="dv">3</span>]], z_scores[fitted<span class="op">$</span>sets<span class="op">$</span>cs[[<span class="dv">3</span>]]],
           fitted<span class="op">$</span>pip[sets<span class="op">$</span>cs[[<span class="dv">3</span>]]])
<span class="co"># Warning in cbind(fitted$sets$cs[[3]], z_scores[fitted$sets$cs[[3]]], fitted</span>
<span class="co"># $pip[sets$cs[[3]]]): number of rows of result is not a multiple of vector</span>
<span class="co"># length (arg 3)</span>
<span class="kw">colnames</span>(z3) =<span class="st"> </span><span class="kw">c</span>(<span class="st">'position'</span>, <span class="st">'z-score'</span>, <span class="st">'PIP'</span>)
z3[<span class="kw">order</span>(z3[,<span class="dv">2</span>], <span class="dt">decreasing =</span> <span class="ot">TRUE</span>),]
<span class="co">#       position  z-score        PIP</span>
<span class="co">#  [1,]      396 5.189811 0.02155638</span>
<span class="co">#  [2,]      381 5.164794 0.05451671</span>
<span class="co">#  [3,]      386 5.164794 0.04081718</span>
<span class="co">#  [4,]      379 5.077563 0.05451671</span>
<span class="co">#  [5,]      391 5.068388 0.05669253</span>
<span class="co">#  [6,]      383 5.057053 0.09412153</span>
<span class="co">#  [7,]      384 5.057053 0.04221480</span>
<span class="co">#  [8,]      389 5.052519 0.01980630</span>
<span class="co">#  [9,]      405 5.039617 0.03075121</span>
<span class="co"># [10,]      403 5.035949 0.04528843</span>
<span class="co"># [11,]      387 5.013526 0.04320035</span>
<span class="co"># [12,]      388 4.997955 0.05738339</span>
<span class="co"># [13,]      408 4.994865 0.09412153</span>
<span class="co"># [14,]      404 4.954407 0.02349161</span>
<span class="co"># [15,]      374 4.948060 0.09412153</span>
<span class="co"># [16,]      373 4.934410 0.05403496</span>
<span class="co"># [17,]      362 4.894243 0.02349161</span>
<span class="co"># [18,]      399 4.860780 0.03376318</span>
<span class="co"># [19,]      392 4.856384 0.02636546</span>
<span class="co"># [20,]      407 4.849285 0.05403496</span>
<span class="co"># [21,]      400 4.827361 0.04883430</span>
<span class="co"># [22,]      398 4.751205 0.03302799</span>
<span class="co"># [23,]      401 4.723184 0.01535169</span>
<span class="co"># [24,]      397 4.716886 0.01493341</span>
<span class="co"># [25,]      415 4.663208 0.05451671</span>
<span class="co"># [26,]      372 4.581560 0.03075121</span></code></pre></div>
</div>
<div id="choice-of-priors" class="section level3">
<h3 class="hasAnchor">
<a href="#choice-of-priors" class="anchor"></a>Choice of priors</h3>
<p>We found that SuSiE is generally robust to choice of priors. Here we set scaled prior variance (percentage of variance explained) to 0.2, and compare with previous results:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitted =<span class="st"> </span><span class="kw"><a href="../reference/susie.html">susie</a></span>(dat<span class="op">$</span>data<span class="op">$</span>X, dat<span class="op">$</span>data<span class="op">$</span>Y[,<span class="dv">1</span>],
               <span class="dt">L=</span><span class="dv">5</span>,
               <span class="dt">estimate_residual_variance =</span> <span class="ot">TRUE</span>, 
               <span class="dt">scaled_prior_variance =</span> <span class="fl">0.2</span>)
<span class="kw"><a href="../reference/susie_plot.html">susie_plot</a></span>(fitted, <span class="dt">y=</span><span class="st">'PIP'</span>, <span class="dt">b=</span>b)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_files/figure-html/unnamed-chunk-14-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>which largely remains unchanged.</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<p>Here are some details about the computing environment, including the versions of R, and the R packages, used to generate these results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co"># R version 3.4.3 (2017-11-30)</span>
<span class="co"># Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co"># Running under: macOS High Sierra 10.13.6</span>
<span class="co"># </span>
<span class="co"># Matrix products: default</span>
<span class="co"># BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib</span>
<span class="co"># LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib</span>
<span class="co"># </span>
<span class="co"># locale:</span>
<span class="co"># [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co"># </span>
<span class="co"># attached base packages:</span>
<span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co"># </span>
<span class="co"># other attached packages:</span>
<span class="co"># [1] susieR_0.4.12</span>
<span class="co"># </span>
<span class="co"># loaded via a namespace (and not attached):</span>
<span class="co">#  [1] Rcpp_0.12.18       matrixStats_0.54.0 lattice_0.20-35   </span>
<span class="co">#  [4] digest_0.6.16      rprojroot_1.3-2    grid_3.4.3        </span>
<span class="co">#  [7] backports_1.1.2    magrittr_1.5       evaluate_0.10.1   </span>
<span class="co"># [10] highr_0.6          stringi_1.1.7      Matrix_1.2-12     </span>
<span class="co"># [13] rmarkdown_1.9      tools_3.4.3        stringr_1.3.0     </span>
<span class="co"># [16] yaml_2.2.0         compiler_3.4.3     htmltools_0.3.6   </span>
<span class="co"># [19] knitr_1.20</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#the-data-set">The data-set</a></li>
      <li><a href="#simple-regression-summary-statistics">Simple regression summary statistics</a></li>
      <li><a href="#fine-mapping-with-susier">Fine-mapping with <code>susieR</code></a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Matthew Stephens, Yuxin Zou, Kaiqian Zhang, Peter Carbonetto, Gao Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
