<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Fine-mapping examples • susieR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Fine-mapping examples">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">susieR</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.13</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stephenslab/susieR">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Fine-mapping examples</h1>
                        <h4 class="author">Gao Wang</h4>
            
            <h4 class="date">June 26, 2018</h4>
      
      
      <div class="hidden name"><code>finemapping.Rmd</code></div>

    </div>

    
    
<div id="fine-mapping-with-susier-and-others" class="section level1">
<h1 class="hasAnchor">
<a href="#fine-mapping-with-susier-and-others" class="anchor"></a>Fine-mapping with <code>susieR</code> and others</h1>
<p>This vignettes demonstrates <code>susieR</code> in the context of genetic fine-mapping. We use simulated data of expression level of a gene (<span class="math inline">\(y\)</span>) in <span class="math inline">\(N \approx 600\)</span> individuals. We want to identify with the genotype matrix <span class="math inline">\(X_{N\times P}\)</span> (<span class="math inline">\(P=1000\)</span>) the genetic variables that causes changes in expression level.</p>
<p>The simulated data-set is <a href="https://github.com/stephenslab/susieR/blob/master/inst/data/N3finemapping.rds">available here</a>, as part of the <code>susieR</code> package. It is a simulate to have exactly 3 non-zero effects.</p>
<div id="the-data-set" class="section level2">
<h2 class="hasAnchor">
<a href="#the-data-set" class="anchor"></a>The data-set</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">dat =<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">system.file</span>(<span class="st">"data"</span>, <span class="st">"N3finemapping.rds"</span>, <span class="dt">package =</span> <span class="st">"susieR"</span>))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">names</span>(dat)</a></code></pre></div>
<pre><code>## [1] "data"     "sumstats"</code></pre>
<p><code>data</code> contains regression data-set <span class="math inline">\(X\)</span> and <span class="math inline">\(y\)</span>, along with some other relevant properties in the context of genetic studies. It also contains the “true” regression coefficent the data is simulated from.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">names</span>(dat<span class="op">$</span>data)</a></code></pre></div>
<pre><code>## [1] "X"                 "chrom"             "pos"              
## [4] "true_coef"         "residual_variance" "Y"                
## [7] "allele_freq"       "V"</code></pre>
<p>Notice that we’ve simulated 2 sets of <span class="math inline">\(Y\)</span> as 2 simulation replicates. Here we’ll focus on the first data-set.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">dim</span>(dat<span class="op">$</span>data<span class="op">$</span>Y)</a></code></pre></div>
<pre><code>## [1] 574   2</code></pre>
<p>Here are the 3 “true” signals in the first data-set:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">plot</span>(dat<span class="op">$</span>data<span class="op">$</span>true_coef[,<span class="dv">1</span>], <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">ylab=</span><span class="st">'effect size'</span>)</a></code></pre></div>
<p><img src="finemapping_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">signal_index =<span class="st"> </span><span class="kw">which</span>(dat<span class="op">$</span>data<span class="op">$</span>true_coef[,<span class="dv">1</span>] <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw">print</span>(signal_index)</a></code></pre></div>
<pre><code>## [1] 403 653 773</code></pre>
<p>So the underlying causal variables are 403, 653 and 773.</p>
</div>
<div id="simple-regression-summary-statistics" class="section level2">
<h2 class="hasAnchor">
<a href="#simple-regression-summary-statistics" class="anchor"></a>Simple regression summary statistics</h2>
<p>The data-set additionally provides summary statistics for fitting univariate simple regression variable by variable. The results are <span class="math inline">\(\hat{\beta}\)</span> and <span class="math inline">\(SE(\hat{\beta})\)</span> from which z-scores can be derived. Again we focus only on results from the first data-set:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">z_scores =<span class="st"> </span>dat<span class="op">$</span>sumstats[<span class="dv">1</span>,,] <span class="op">/</span><span class="st"> </span>dat<span class="op">$</span>sumstats[<span class="dv">2</span>,,]</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">z_scores =<span class="st"> </span>z_scores[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="kw">plot</span>(z_scores, <span class="dt">pch=</span><span class="dv">16</span>)</a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="kw">points</span>(signal_index, z_scores[signal_index], <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">pch =</span> <span class="dv">16</span>)</a></code></pre></div>
<p><img src="finemapping_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div id="fine-mapping-with-susier" class="section level2">
<h2 class="hasAnchor">
<a href="#fine-mapping-with-susier" class="anchor"></a>Fine-mapping with <code>susieR</code>
</h2>
<p>For starters, we assume there are at most 5 causal variables, i.e., set <code>L=5</code>, although SuSiE is robust to the choice of <code>L</code>.</p>
<p>We update residual variance as well as prior variance in the variational algorithm that fits SuSiE model. The <code>susieR</code> function call is:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">fitted =<span class="st"> </span>susieR<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/susieR/topics/susie">susie</a></span>(dat<span class="op">$</span>data<span class="op">$</span>X, dat<span class="op">$</span>data<span class="op">$</span>Y[,<span class="dv">1</span>],</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">                       <span class="dt">L=</span><span class="dv">5</span>,</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">                       <span class="dt">estimate_residual_variance =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb11-4" data-line-number="4">                       <span class="dt">estimate_prior_variance =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb11-5" data-line-number="5">                       <span class="dt">intercept=</span><span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">                       <span class="dt">tol=</span><span class="fl">1e-3</span>)</a></code></pre></div>
<div id="confidence-sets" class="section level3">
<h3 class="hasAnchor">
<a href="#confidence-sets" class="anchor"></a>Confidence sets</h3>
<p>We obtain 95% confidence set by:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">sets =<span class="st"> </span>susieR<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/susieR/topics/susie_get_CS">susie_get_CS</a></span>(fitted,</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">                            <span class="dt">coverage =</span> <span class="fl">0.95</span>,</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">                            <span class="dt">X=</span> dat<span class="op">$</span>data<span class="op">$</span>X, </a>
<a class="sourceLine" id="cb12-4" data-line-number="4">                            <span class="dt">min_abs_corr =</span> <span class="fl">0.1</span>)</a></code></pre></div>
<p>Notice that by passing to <code>susie_get_CS</code> function the data matrix <code>X</code>, we further filter the sets identified by removing those having absolute correlation between any pairs of variables in the set greater than specified value, which is 0.1 in the function call above. We call this a “purity” filter.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">print</span>(sets)</a></code></pre></div>
<pre><code>## $cs
## $cs[[1]]
## [1] 773 777
## 
## $cs[[2]]
## [1] 653
## 
## $cs[[3]]
##  [1] 362 365 372 373 374 379 381 383 384 386 387 388 389 391 392 396 397
## [18] 398 399 400 401 403 404 405 407 408 415
## 
## 
## $purity
##   min.abs.corr mean.abs.corr median.abs.corr
## 1    0.9815726     0.9907863       0.9907863
## 2    1.0000000     1.0000000       1.0000000
## 3    0.8686309     0.9653503       0.9739578
## 
## $cs_index
## [1] 1 2 3</code></pre>
<p>The 3 causal signals have been captured by the 3 CS reported here. The 3rd CS contains many variables, including the true causal variable <code>403</code>. The minimum absolute correlation is 0.86.</p>
<p>If we use the default 90% coverage for confidence sets, we still capture the 3 signals, but “purity” of the 3rd CS is now 0.91 and size of the CS is also a bit smaller.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">sets =<span class="st"> </span>susieR<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/susieR/topics/susie_get_CS">susie_get_CS</a></span>(fitted,</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">                            <span class="dt">X=</span> dat<span class="op">$</span>data<span class="op">$</span>X, </a>
<a class="sourceLine" id="cb15-3" data-line-number="3">                            <span class="dt">min_abs_corr =</span> <span class="fl">0.1</span>)</a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">print</span>(sets)</a></code></pre></div>
<pre><code>## $cs
## $cs[[1]]
## [1] 773 777
## 
## $cs[[2]]
## [1] 653
## 
## $cs[[3]]
##  [1] 373 374 379 381 383 384 386 387 388 389 391 392 396 398 399 400 401
## [18] 403 404 405 407 408
## 
## 
## $purity
##   min.abs.corr mean.abs.corr median.abs.corr
## 1    0.9815726     0.9907863       0.9907863
## 2    1.0000000     1.0000000       1.0000000
## 3    0.9119572     0.9738724       0.9768163
## 
## $cs_index
## [1] 1 2 3</code></pre>
</div>
</div>
<div id="posterior-inclusion-probability" class="section level2">
<h2 class="hasAnchor">
<a href="#posterior-inclusion-probability" class="anchor"></a>Posterior inclusion probability</h2>
<p>Previously we’ve determined that summing over 3 single effect regression models is approperate for our application. Here we summarize the variable selection results by computing posterior inclusion probability for SuSiE model:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">pip =<span class="st"> </span>susieR<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/susieR/topics/susie_get_PIP">susie_get_PIP</a></span>(fitted, sets<span class="op">$</span>cs_index)</a></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">plot</span>(pip, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">ylab=</span><span class="st">'effect size'</span>)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="kw">points</span>(signal_index, pip[signal_index], <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">pch =</span> <span class="dv">16</span>)</a></code></pre></div>
<p><img src="finemapping_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>The true causal variables are colored red. Of interest is the cluster around position 400. The true signal is 403 but apparently it does not have the highest PIP. This is consistent with its ranking in original z-score in that CS:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">z3 =<span class="st"> </span><span class="kw">cbind</span>(sets<span class="op">$</span>cs[[<span class="dv">3</span>]], z_scores[sets<span class="op">$</span>cs[[<span class="dv">3</span>]]])</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">z3[<span class="kw">order</span>(z3[,<span class="dv">2</span>], <span class="dt">decreasing =</span> <span class="ot">TRUE</span>),]</a></code></pre></div>
<pre><code>##       [,1]     [,2]
##  [1,]  396 5.189811
##  [2,]  381 5.164794
##  [3,]  386 5.164794
##  [4,]  379 5.077563
##  [5,]  391 5.068388
##  [6,]  383 5.057053
##  [7,]  384 5.057053
##  [8,]  389 5.052519
##  [9,]  405 5.039617
## [10,]  403 5.035949
## [11,]  387 5.013526
## [12,]  388 4.997955
## [13,]  408 4.994865
## [14,]  404 4.954407
## [15,]  374 4.948060
## [16,]  373 4.934410
## [17,]  399 4.860780
## [18,]  392 4.856384
## [19,]  407 4.849285
## [20,]  400 4.827361
## [21,]  398 4.751205
## [22,]  401 4.723184</code></pre>
</div>
<div id="fine-mapping-with-other-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#fine-mapping-with-other-methods" class="anchor"></a>Fine-mapping with other methods</h2>
<p>In <a href="https://github.com/stephenslab/susieR/tree/master/inst/code">this folder</a> we provide scripts to run <code>CAVIAR</code>, <code>FINEMAP</code> (version 1.1) and <code>DAP-G</code> on the same data-set for a comparison. The scripts along with software dependencies are also available from the docker image for <code>susieR</code> demonstrations. To run from the docker image <code>gaow/susie</code>, first <a href="https://www.docker.com/community-edition">install docker</a>, then set command-line alias <code>docker-susie</code>:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="bu">alias</span> docker-susie=<span class="st">"docker run --rm --security-opt label:disable -t -h susie -P -w </span><span class="va">$PWD</span><span class="st"> -v </span><span class="va">$PWD</span><span class="st">:</span><span class="va">$PWD</span><span class="st"> -u </span><span class="va">$UID</span><span class="st">:</span><span class="va">${GROUPS[0]}</span><span class="st"> -e HOME=/home/</span><span class="va">$USER</span><span class="st"> -e USER=</span><span class="va">$USER</span><span class="st"> gaow/susie"</span></a></code></pre></div>
<p>Now you are set to run the aforementioned fine-mapping programs.</p>
<div id="finemap" class="section level3">
<h3 class="hasAnchor">
<a href="#finemap" class="anchor"></a>FINEMAP</h3>
<p>We use default <code>FINEMAP</code> priors, and set maximum number of causal variables to 3.</p>
<pre><code>docker-susie finemap.R input=\"N3finemapping.rds\" output=\"N3finemapping.FINEMAP\" args=\"--n-causal-max\ 3\"</code></pre>
</div>
<div id="caviar" class="section level3">
<h3 class="hasAnchor">
<a href="#caviar" class="anchor"></a>CAVIAR</h3>
<p>We set <code>CAVIAR</code> prior probability of a variable being causal to 1 / P, and set maximum number of causal variables to 3.</p>
<pre><code>docker-susie caviar.R input=\"N3finemapping.rds\" output=\"N3finemapping.CAVIAR\" args=\"-g\ 0.001\ -c\ 3\"</code></pre>
</div>
<div id="dap-g" class="section level3">
<h3 class="hasAnchor">
<a href="#dap-g" class="anchor"></a>DAP-G</h3>
<p>We use full data vesion of DAP-G, that is, the input is the data matrices not summary statistics.</p>
<pre><code>docker-susie dap-g.py N3finemapping.rds N3finemapping.DAP -ld_control 0.20 --all</code></pre>
<p>For convenience we include these results as part of the <code>susieR</code> package data files:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">caviar =<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">system.file</span>(<span class="st">"data"</span>, <span class="st">"N3finemapping.CAVIAR.rds"</span>, <span class="dt">package =</span> <span class="st">"susieR"</span>))</a>
<a class="sourceLine" id="cb26-2" data-line-number="2">finemap =<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">system.file</span>(<span class="st">"data"</span>, <span class="st">"N3finemapping.FINEMAP.rds"</span>, <span class="dt">package =</span> <span class="st">"susieR"</span>))</a>
<a class="sourceLine" id="cb26-3" data-line-number="3">dap =<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">system.file</span>(<span class="st">"data"</span>, <span class="st">"N3finemapping.DAP.rds"</span>, <span class="dt">package =</span> <span class="st">"susieR"</span>))</a></code></pre></div>
</div>
<div id="caviar-1" class="section level3">
<h3 class="hasAnchor">
<a href="#caviar-1" class="anchor"></a>CAVIAR</h3>
<div id="pip" class="section level4">
<h4 class="hasAnchor">
<a href="#pip" class="anchor"></a>PIP</h4>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">snp =<span class="st"> </span>caviar[[<span class="dv">1</span>]]<span class="op">$</span>snp</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">pip =<span class="st"> </span>snp[<span class="kw">order</span>(<span class="kw">as.numeric</span>(snp<span class="op">$</span>snp)),]<span class="op">$</span>snp_prob</a>
<a class="sourceLine" id="cb27-3" data-line-number="3"><span class="kw">plot</span>(pip, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">ylab=</span><span class="st">'effect size'</span>)</a>
<a class="sourceLine" id="cb27-4" data-line-number="4"><span class="kw">points</span>(signal_index, pip[signal_index], <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">pch =</span> <span class="dv">16</span>)</a></code></pre></div>
<p><img src="finemapping_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
<div id="cs" class="section level4">
<h4 class="hasAnchor">
<a href="#cs" class="anchor"></a>95% CS</h4>
<p>CAVIAR provides single 95% confidence sets as follows:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw">print</span>(caviar[[<span class="dv">1</span>]]<span class="op">$</span>set)</a></code></pre></div>
<pre><code>##  [1] "653" "773" "777" "381" "386" "396" "391" "379" "383" "384" "405"
## [12] "389" "387" "408" "388" "403" "404" "374" "373" "399" "400" "392"
## [23] "407" "362" "401" "398" "397" "415" "365" "368" "366"</code></pre>
<p>which captures positions 653, 773 and 403.</p>
</div>
</div>
<div id="finemap-1" class="section level3">
<h3 class="hasAnchor">
<a href="#finemap-1" class="anchor"></a>FINEMAP</h3>
<div id="pip-1" class="section level4">
<h4 class="hasAnchor">
<a href="#pip-1" class="anchor"></a>PIP</h4>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1">snp =<span class="st"> </span>finemap[[<span class="dv">1</span>]]<span class="op">$</span>snp</a>
<a class="sourceLine" id="cb30-2" data-line-number="2">pip =<span class="st"> </span>snp[<span class="kw">order</span>(<span class="kw">as.numeric</span>(snp<span class="op">$</span>snp)),]<span class="op">$</span>snp_prob</a>
<a class="sourceLine" id="cb30-3" data-line-number="3"><span class="kw">plot</span>(pip, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">ylab=</span><span class="st">'effect size'</span>)</a>
<a class="sourceLine" id="cb30-4" data-line-number="4"><span class="kw">points</span>(signal_index, pip[signal_index], <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">pch =</span> <span class="dv">16</span>)</a></code></pre></div>
<p><img src="finemapping_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
</div>
<div id="cs-1" class="section level4">
<h4 class="hasAnchor">
<a href="#cs-1" class="anchor"></a>95% CS</h4>
<p>For each configuration, FINEMAP provides the corresponding configuration probability:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="kw">head</span>(finemap[[<span class="dv">1</span>]]<span class="op">$</span>set, <span class="dv">10</span>)</a></code></pre></div>
<pre><code>##    rank      config config_prob config_log10bf config_prob_cumsum
## 1     1 386,653,773  0.02996234       15.24259         0.02996234
## 2     2 381,653,773  0.02996234       15.24259         0.05992468
## 3     3     653,773  0.02985153       12.24098         0.08977621
## 4     4 396,653,773  0.02778146       15.20977         0.11755767
## 5     5 391,653,773  0.02432951       15.15215         0.14188718
## 6     6 379,653,773  0.02429567       15.15154         0.16618285
## 7     7 384,653,773  0.02357643       15.13849         0.18975929
## 8     8 383,653,773  0.02357643       15.13849         0.21333572
## 9     9 405,653,773  0.02201540       15.10874         0.23535111
## 10   10 389,653,773  0.02167742       15.10202         0.25702853</code></pre>
<p>The top ranking configurations correctly captures position 653 and 773, but not 403. It requires post-processing to come up with a single 95% set similar to that of CAVIAR with all 3 causal variables captured. The result from running <code>finemap.R</code> provided has been truncated to the minimum set of configurations having cumulative probability of 95%.</p>
</div>
</div>
<div id="dap-g-1" class="section level3">
<h3 class="hasAnchor">
<a href="#dap-g-1" class="anchor"></a>DAP-G</h3>
<div id="pip-2" class="section level4">
<h4 class="hasAnchor">
<a href="#pip-2" class="anchor"></a>PIP</h4>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1">snp =<span class="st"> </span>dap[[<span class="dv">1</span>]]<span class="op">$</span>snp</a>
<a class="sourceLine" id="cb33-2" data-line-number="2">pip =<span class="st"> </span>snp[<span class="kw">order</span>(<span class="kw">as.numeric</span>(snp<span class="op">$</span>snp)),]<span class="op">$</span>snp_prob</a>
<a class="sourceLine" id="cb33-3" data-line-number="3"><span class="kw">plot</span>(pip, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">ylab=</span><span class="st">'effect size'</span>)</a>
<a class="sourceLine" id="cb33-4" data-line-number="4"><span class="kw">points</span>(signal_index, pip[signal_index], <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">pch =</span> <span class="dv">16</span>)</a></code></pre></div>
<p><img src="finemapping_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
</div>
<div id="cs-2" class="section level4">
<h4 class="hasAnchor">
<a href="#cs-2" class="anchor"></a>95% CS</h4>
<p>Similar to SuSiE, DAP-G provides per-signal 95% CS:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">dap[[<span class="dv">1</span>]]<span class="op">$</span>set</a></code></pre></div>
<pre><code>##   cluster cluster_prob cluster_avg_r2
## 0       1    0.9999000          0.963
## 1       2    0.9999000          1.000
## 2       3    0.9994000          0.895
## 3       4    0.0522200          0.369
## 4       5    0.0003854          0.863
##                                                                                                                                                               snp
## 0                                                                                                                                                         773,777
## 1                                                                                                                                                             653
## 2 386,381,396,391,379,383,384,405,408,389,387,388,404,403,374,399,373,400,392,401,407,362,398,415,397,365,372,368,360,366,367,361,419,410,409,349,369,424,402,347
## 3                                             584,599,472,565,614,569,571,766,561,578,582,568,585,640,193,607,608,617,847,586,863,873,219,208,212,865,866,880,197
## 4                                                                                                                                                         747,771</code></pre>
<p>Using an average <span class="math inline">\(r^2\)</span> filter of 0.2 (<span class="math inline">\(r=0.44\)</span> compared to SuSiE’s <span class="math inline">\(r=0.1\)</span> in this analysis), it reports 5 95% CS. But only the first 3 CS contain causal signals.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#fine-mapping-with-susier-and-others">Fine-mapping with <code>susieR</code> and others</a><ul class="nav nav-pills nav-stacked">
<li><a href="#the-data-set">The data-set</a></li>
      <li><a href="#simple-regression-summary-statistics">Simple regression summary statistics</a></li>
      <li><a href="#fine-mapping-with-susier">Fine-mapping with <code>susieR</code></a></li>
      <li><a href="#posterior-inclusion-probability">Posterior inclusion probability</a></li>
      <li><a href="#fine-mapping-with-other-methods">Fine-mapping with other methods</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Matthew Stephens, Yuxin Zou, Peter Carbonetto, Gao Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
