<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Fine-mapping with summary statistics • susieR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Fine-mapping with summary statistics">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">susieR</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.5.0.350</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stephenslab/susieR">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Fine-mapping with summary statistics</h1>
                        <h4 class="author">Yuxin Zou and Gao Wang</h4>
            
            <h4 class="date">2018-10-19</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/stephenslab/susieR/blob/master/vignettes/finemapping_summary_statistics.Rmd"><code>vignettes/finemapping_summary_statistics.Rmd</code></a></small>
      <div class="hidden name"><code>finemapping_summary_statistics.Rmd</code></div>

    </div>

    
    
<p>This vignette demonstrates how to use <code>susieR</code> with “summary statistics” in the context of genetic fine-mapping. We use the same simulated data as in <a href="finemapping.html">fine mapping vignette</a>. The simulated data is expression level of a gene (<span class="math inline">\(y\)</span>) in <span class="math inline">\(N \approx 600\)</span> individuals. We want to identify with the genotype matrix <span class="math inline">\(X_{N\times P}\)</span> (<span class="math inline">\(P=1000\)</span>) the genetic variables that causes changes in expression level. The data-set is shipped with <code>susieR</code>. It is simulated to have exactly 3 non-zero effects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(susieR)
<span class="kw">set.seed</span>(<span class="dv">1</span>)</code></pre></div>
<div id="the-data-set" class="section level2">
<h2 class="hasAnchor">
<a href="#the-data-set" class="anchor"></a>The data-set</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(N3finemapping)
<span class="kw">attach</span>(N3finemapping)</code></pre></div>
<p>Notice that we’ve simulated 2 sets of <span class="math inline">\(Y\)</span> as 2 simulation replicates. Here we’ll focus on the first data-set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(data$Y)
<span class="co"># [1] 574   2</span></code></pre></div>
<p>Here are the 3 “true” signals in the first data-set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">b &lt;-<span class="st"> </span>data$true_coef[,<span class="dv">1</span>]
<span class="kw">plot</span>(b, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">ylab=</span><span class="st">'effect size'</span>)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-4-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">which</span>(b !=<span class="st"> </span><span class="dv">0</span>)
<span class="co"># [1] 403 653 773</span></code></pre></div>
<p>So the underlying causal variables are 403, 653 and 773.</p>
</div>
<div id="summary-statistics-from-simple-regression" class="section level2">
<h2 class="hasAnchor">
<a href="#summary-statistics-from-simple-regression" class="anchor"></a>Summary statistics from simple regression</h2>
<p>Summary statistics of genetic association studies typically contain odds ratio (or <span class="math inline">\(\hat{\beta}\)</span> coefficient), p-value and minor allele frequencies. These information can be used to perform fine-mapping with sufficient summary statistics, which additionally require sample size, variance of phenotype, and most importantly a matrix of correlation between variables.</p>
<p>When the aforementioned sufficient summary statistica are provided, SuSiE can produce the exact same outcome as if individual level data were used. SuSiE can also produce approximate results when some of the information are not available. The minimal required of summary statistics for SuSiE to work are z-scores (can be derived from p-values, as will be demonstrated below) and correlation matrix between variables. The correlation matrix in genetics is typically referred to as LD matrix (LD for linkage disequilibrium). One can use external reference panels to estimate it when this matrix cannot be obtained from samples directly.</p>
<p>Caution that the LD matrix has to be correlation matrix <span class="math inline">\(r\)</span>, not <span class="math inline">\(r^2\)</span> or <span class="math inline">\(abs(r)\)</span>.</p>
<p>Our example data-set additionally provides summary statistics for fitting per-variable univariate simple regression. The results are <span class="math inline">\(\hat{\beta}\)</span> and <span class="math inline">\(SE(\hat{\beta})\)</span> from which the p-values (from <em>t</em>-distribution) and <span class="math inline">\(z\)</span>-scores can be derived. Again we focus only on results from the first data-set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">p_values =<span class="st"> </span><span class="dv">2</span> *<span class="st"> </span><span class="kw">pt</span>(-<span class="kw">abs</span>(sumstats[<span class="dv">1</span>,,<span class="dv">1</span>] /<span class="st"> </span>sumstats[<span class="dv">2</span>,,<span class="dv">1</span>]), <span class="dt">df =</span> <span class="kw">nrow</span>(data$X) -<span class="st"> </span><span class="dv">2</span>)
z_scores =<span class="st"> </span><span class="kw">abs</span>(<span class="kw">qnorm</span>(p_values/<span class="dv">2</span>)) *<span class="st"> </span><span class="kw">sign</span>(sumstats[<span class="dv">1</span>,,<span class="dv">1</span>])
<span class="kw"><a href="../reference/susie_plot.html">susie_plot</a></span>(z_scores, <span class="dt">y =</span> <span class="st">"z"</span>, <span class="dt">b=</span>b)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-6-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>For this example the correlation matrix can be computed directly from data provide,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">R &lt;-<span class="st"> </span><span class="kw">cor</span>(data$X)</code></pre></div>
</div>
<div id="fine-mapping-with-susier-with-the-knowledge-of-sample-size-n" class="section level2">
<h2 class="hasAnchor">
<a href="#fine-mapping-with-susier-with-the-knowledge-of-sample-size-n" class="anchor"></a>Fine-mapping with <code>susieR</code> with the knowledge of sample size n</h2>
<p>For starters, we assume there are at most 5 causal variables, i.e. set <code>L=5</code>, although SuSiE is robust to the choice of <code>L</code>.</p>
<p>We also set SuSiE prior variance to 0.1.</p>
<div id="using-hatbeta-and-textsehatbeta" class="section level3">
<h3 class="hasAnchor">
<a href="#using-hatbeta-and-textsehatbeta" class="anchor"></a>Using <span class="math inline">\(\hat{\beta}\)</span> and <span class="math inline">\(\text{SE}(\hat{\beta})\)</span>
</h3>
<p>With the knowledge of variance of <span class="math inline">\(y\)</span>, <code>susieR</code> can result in coefficent estimates in the original scale of data,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitted_bhat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/susie_bhat.html">susie_bhat</a></span>(<span class="dt">bhat =</span> sumstats[<span class="dv">1</span>,,<span class="dv">1</span>], 
                          <span class="dt">shat =</span> sumstats[<span class="dv">2</span>,,<span class="dv">1</span>], 
                          <span class="dt">R =</span> R, <span class="dt">n =</span> <span class="kw">nrow</span>(data$X), 
                          <span class="dt">var_y =</span> <span class="kw">var</span>(data$Y[,<span class="dv">1</span>]),
                          <span class="dt">L =</span> <span class="dv">5</span>, 
                          <span class="dt">scaled_prior_variance =</span> <span class="fl">0.1</span>, 
                          <span class="dt">estimate_prior_variance =</span> <span class="ot">FALSE</span>, 
                          <span class="dt">standardize =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Without the knowledge of variance of <span class="math inline">\(y\)</span>, <code>susieR</code> gives the coefficients in standardized <span class="math inline">\(X\)</span>, <span class="math inline">\(y\)</span> scale,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitted_bhat_standardize &lt;-<span class="st"> </span><span class="kw"><a href="../reference/susie_bhat.html">susie_bhat</a></span>(<span class="dt">bhat =</span> sumstats[<span class="dv">1</span>,,<span class="dv">1</span>], 
                                      <span class="dt">shat =</span> sumstats[<span class="dv">2</span>,,<span class="dv">1</span>], 
                                      <span class="dt">R =</span> R, <span class="dt">n =</span> <span class="kw">nrow</span>(data$X), 
                                      <span class="dt">L =</span> <span class="dv">5</span>, 
                                      <span class="dt">scaled_prior_variance =</span> <span class="fl">0.1</span>, 
                                      <span class="dt">estimate_prior_variance =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>Using <code>summary</code> function, we can examine the posterior inclusion probability (PIP) for each variable, and the 95% credible sets.</p>
<p>Here, we are the 95% credible sets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(fitted_bhat)$cs
<span class="co">#   cs cs_log10bf cs_avg_r2 cs_min_r2</span>
<span class="co"># 1  3  12.947239 1.0000000 1.0000000</span>
<span class="co"># 2  1  17.621410 0.9816575 0.9634847</span>
<span class="co"># 3  2   9.874888 0.9114032 0.6907024</span>
<span class="co">#                                                                                                                  variable</span>
<span class="co"># 1                                                                                                                     653</span>
<span class="co"># 2                                                                                                                 773,777</span>
<span class="co"># 3 360,361,362,365,368,372,373,374,379,381,383,384,386,387,388,389,391,392,396,397,398,399,400,401,403,404,405,407,408,415</span></code></pre></div>
<p>The 3 causal signals have been captured by the 3 CS reported here. The 3rd CS contains many variables, including the true causal variable 403.</p>
<p>We can also plot the posterior inclusion probability (PIP),</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/susie_plot.html">susie_plot</a></span>(fitted_bhat, <span class="dt">y=</span><span class="st">"PIP"</span>, <span class="dt">b=</span>b)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-11-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>The true causal variables are colored red. The 95% CS identified are circled in different colors.</p>
<p>The result should be identical to using the individual level data,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitted =<span class="st"> </span><span class="kw"><a href="../reference/susie.html">susie</a></span>(data$X, data$Y[,<span class="dv">1</span>],
                <span class="dt">L=</span><span class="dv">5</span>,
                <span class="dt">estimate_residual_variance =</span> <span class="ot">TRUE</span>, 
                <span class="dt">scaled_prior_variance =</span> <span class="fl">0.1</span>,
                <span class="dt">verbose =</span> <span class="ot">TRUE</span>)
<span class="co"># [1] "objective:-1384.57253292456"</span>
<span class="co"># [1] "objective:-1381.26163424408"</span>
<span class="co"># [1] "objective:-1380.75139979919"</span>
<span class="co"># [1] "objective:-1380.74854534135"</span>
<span class="co"># [1] "objective:-1380.6759533616"</span>
<span class="co"># [1] "objective:-1380.67587727196"</span>
<span class="co"># [1] "objective:-1377.04794661565"</span>
<span class="co"># [1] "objective:-1376.93001122693"</span>
<span class="co"># [1] "objective:-1373.29122910603"</span>
<span class="co"># [1] "objective:-1373.25708807704"</span>
<span class="co"># [1] "objective:-1373.20171636119"</span>
<span class="co"># [1] "objective:-1373.20167955938"</span>
<span class="co"># [1] "objective:-1373.20062377817"</span>
<span class="co"># [1] "objective:-1373.20062375159"</span>
<span class="co"># [1] "objective:-1373.20061073365"</span>
<span class="co"># [1] "objective:-1373.20061072967"</span>
<span class="kw">plot</span>(fitted$pip, fitted_bhat$pip)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-12-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">coef</span>(fitted), <span class="kw">coef</span>(fitted_bhat_standardize))</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-13-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
</div>
<div id="using-z-scores" class="section level3">
<h3 class="hasAnchor">
<a href="#using-z-scores" class="anchor"></a>Using <span class="math inline">\(z\)</span> scores</h3>
<p>For <span class="math inline">\(z\)</span> score input, <code>susieR</code> can only output coefficients in standardized <span class="math inline">\(X\)</span>, <span class="math inline">\(y\)</span> scale. It is practically setting <span class="math inline">\(\hat{\beta}\)</span> to <span class="math inline">\(z\)</span> score and <span class="math inline">\(\text{SE}(\hat{\beta})\)</span> and variance of <span class="math inline">\(y\)</span> to one,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitted_z_n &lt;-<span class="st"> </span><span class="kw"><a href="../reference/susie_bhat.html">susie_bhat</a></span>(<span class="dt">bhat =</span> z_scores, <span class="dt">shat =</span> <span class="dv">1</span>, 
                         <span class="dt">R =</span> R, <span class="dt">n =</span> <span class="kw">nrow</span>(data$X), 
                         <span class="dt">L =</span> <span class="dv">5</span>, 
                         <span class="dt">scaled_prior_variance =</span> <span class="fl">0.1</span>, 
                         <span class="dt">estimate_prior_variance =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>The result should agree with using <span class="math inline">\(\hat{\beta}\)</span> and <span class="math inline">\(\text{SE}(\hat{\beta})\)</span> when variance of <span class="math inline">\(y\)</span> is set to one,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">coef</span>(fitted_z_n), <span class="kw">coef</span>(fitted_bhat_standardize))</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-15-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(fitted_z_n)$cs
<span class="co">#   cs cs_log10bf cs_avg_r2 cs_min_r2</span>
<span class="co"># 1  3  12.389806 1.0000000 1.0000000</span>
<span class="co"># 2  1  16.622625 0.9816575 0.9634847</span>
<span class="co"># 3  2   9.616246 0.9098701 0.6907024</span>
<span class="co">#                                                                                                                      variable</span>
<span class="co"># 1                                                                                                                         653</span>
<span class="co"># 2                                                                                                                     773,777</span>
<span class="co"># 3 360,361,362,365,366,368,372,373,374,379,381,383,384,386,387,388,389,391,392,396,397,398,399,400,401,403,404,405,407,408,415</span></code></pre></div>
<p>and the PIP plot</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/susie_plot.html">susie_plot</a></span>(fitted_bhat, <span class="dt">y=</span><span class="st">"PIP"</span>, <span class="dt">b=</span>b)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-17-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
</div>
</div>
<div id="fine-mapping-with-susier-without-the-knowledge-of-sample-size" class="section level2">
<h2 class="hasAnchor">
<a href="#fine-mapping-with-susier-without-the-knowledge-of-sample-size" class="anchor"></a>Fine-mapping with <code>susieR</code> without the knowledge of sample size</h2>
<p>With only <span class="math inline">\(z\)</span> and <span class="math inline">\(R\)</span> as input, the inference on effects are more conservative. For <span class="math inline">\(z\)</span> score input, <code>susieR</code> estimates prior variance in the iterative variable selection algorithm, and output coefficients in standardized <span class="math inline">\(X\)</span>, <span class="math inline">\(y\)</span> scale,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fitted_z &lt;-<span class="st"> </span><span class="kw"><a href="../reference/susie_z.html">susie_z</a></span>(<span class="dt">z =</span> z_scores,<span class="dt">R =</span> R,<span class="dt">L =</span> <span class="dv">5</span>)</code></pre></div>
<p>We can check the 95% credible set:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(fitted_z)$cs
<span class="co">#   cs cs_log10bf cs_avg_r2 cs_min_r2</span>
<span class="co"># 1  2   13.48473 1.0000000 1.0000000</span>
<span class="co"># 2  1   18.64257 0.9816575 0.9634847</span>
<span class="co"># 3  3   10.21614 0.9170449 0.6907024</span>
<span class="co">#                                                                                                                  variable</span>
<span class="co"># 1                                                                                                                     653</span>
<span class="co"># 2                                                                                                                 773,777</span>
<span class="co"># 3 360,362,365,366,368,372,373,374,379,381,383,384,386,387,388,389,391,392,396,397,398,399,400,401,403,404,405,407,408,415</span></code></pre></div>
<p>and the PIP</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/susie_plot.html">susie_plot</a></span>(fitted_z, <span class="dt">y=</span><span class="st">"PIP"</span>, <span class="dt">b=</span>b)</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-20-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
<p>We compare result with when <span class="math inline">\(n\)</span> is known,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">coef</span>(fitted_z), <span class="kw">coef</span>(fitted_z_n))</code></pre></div>
<div class="figure" style="text-align: center">
<img src="finemapping_summary_statistics_files/figure-html/unnamed-chunk-21-1.png" alt="&amp;nbsp;" width="600"><p class="caption">
 
</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<p>Here are some details about the computing environment, including the versions of R, and the R packages, used to generate these results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co"># R version 3.4.4 (2018-03-15)</span>
<span class="co"># Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co"># Running under: BunsenLabs GNU/Linux 8.7 (Hydrogen)</span>
<span class="co"># </span>
<span class="co"># Matrix products: default</span>
<span class="co"># BLAS: /usr/lib/atlas-base/atlas/libblas.so.3.0</span>
<span class="co"># LAPACK: /usr/lib/atlas-base/atlas/liblapack.so.3.0</span>
<span class="co"># </span>
<span class="co"># locale:</span>
<span class="co">#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co"># [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co"># </span>
<span class="co"># attached base packages:</span>
<span class="co"># [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co"># </span>
<span class="co"># other attached packages:</span>
<span class="co"># [1] susieR_0.5.0.0350</span>
<span class="co"># </span>
<span class="co"># loaded via a namespace (and not attached):</span>
<span class="co">#  [1] Rcpp_0.12.19        knitr_1.20          xml2_1.2.0         </span>
<span class="co">#  [4] magrittr_1.5        roxygen2_6.1.0.9000 MASS_7.3-34        </span>
<span class="co">#  [7] lattice_0.20-29     R6_2.2.2            rlang_0.2.2        </span>
<span class="co"># [10] highr_0.6           stringr_1.3.1       tools_3.4.4        </span>
<span class="co"># [13] grid_3.4.4          matrixStats_0.54.0  htmltools_0.3.6    </span>
<span class="co"># [16] commonmark_1.5      yaml_2.2.0          rprojroot_1.3-2    </span>
<span class="co"># [19] digest_0.6.17       assertthat_0.2.0    pkgdown_1.1.0.9000 </span>
<span class="co"># [22] crayon_1.3.4        Matrix_1.1-4        fs_1.2.6           </span>
<span class="co"># [25] memoise_1.1.0       evaluate_0.10.1     rmarkdown_1.10     </span>
<span class="co"># [28] stringi_1.2.4       compiler_3.4.4      desc_1.2.0         </span>
<span class="co"># [31] backports_1.1.2</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#the-data-set">The data-set</a></li>
      <li><a href="#summary-statistics-from-simple-regression">Summary statistics from simple regression</a></li>
      <li><a href="#fine-mapping-with-susier-with-the-knowledge-of-sample-size-n">Fine-mapping with <code>susieR</code> with the knowledge of sample size n</a></li>
      <li><a href="#fine-mapping-with-susier-without-the-knowledge-of-sample-size">Fine-mapping with <code>susieR</code> without the knowledge of sample size</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Matthew Stephens, Yuxin Zou, Kaiqian Zhang, Peter Carbonetto, Gao Wang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
